driver/src/price_finding/naive_solver.rs

```diff
@@ -0,0 +1,74 @@
+use crate::models;
+
+use super::error::PriceFindingError;
+use super::price_finder_interface::Solution;
+use web3::types::U256;
+use crate::models::Order;
+
+impl Order {
+    fn matches(&self, other: &Order) -> bool {
+        self.opposite_tokens(other) && self.have_price_overlap(other)
+    }
+
+    fn opposite_tokens(&self, other: &Order) -> bool {
+        self.buy_token == other.sell_token && self.sell_token == other.buy_token
+    }
+
+    fn have_price_overlap(&self, other: &Order) -> bool {
+        self.buy_amount * other.buy_amount <= other.sell_amount * self.sell_amount
+    }
+}
+
+pub fn solve(orders: &Vec<models::Order>, num_tokens: u8) -> Result<Solution, PriceFindingError> {
+    // TODO - use mapping here
+    let mut prices: Vec<u128> = vec![0; num_tokens as usize];
+    let mut sell_amount: Vec<u128> = vec![0; orders.len()];
+    let mut buy_amount: Vec<u128> = vec![0; orders.len()];
+
+    for (i, x) in orders.iter().enumerate() {
```

Comment:
A slightly more functional way of writing this could use [combinations](https://docs.rs/itertools/0.7.6/itertools/trait.Itertools.html#method.tuple_combinations) and 

```
let match = orders.tuple_combinations().iter().find(|(o1, o2)| o1.matches(o2).is_some());
```

Reply:
Yes, I was looking for something like this, but hadn't yet found anything. Thanks!

Reply:
This is still a great idea to be implemented in the future.